Sources/DeepLearning/Layer.swift

```diff
@@ -1295,3 +1295,70 @@ public struct Reshape<Scalar: TensorFlowFloatingPoint>: Layer {
         return input.reshaped(toShape: shape)
     }
 }
+
+/// An LSTM Cell.
+public struct LSTMCell<Scalar: TensorFlowFloatingPoint>: Layer {
+    public var inputWeight: Tensor<Scalar>
+    public var updateWeight: Tensor<Scalar>
+    public var forgetWeight: Tensor<Scalar>
+    public var forgetBias: Tensor<Scalar>
+    public var outputWeight: Tensor<Scalar>
+
+    /// Creates a `LSTMCell` with the specified input size and hidden state size.
+    ///
+    /// - Parameters:
+    ///   - inputSize: The number of features in 2-D input tensors.
+    ///   - hiddenSize: The number of features in 2-D hidden states.
+    public init(inputSize: Int32, hiddenSize: Int32) {
+        let concatenatedInputSize = inputSize + hiddenSize
+        self.inputWeight = Tensor<Scalar>(glorotUniform: [concatenatedInputSize, hiddenSize])
+        self.updateWeight = Tensor<Scalar>(glorotUniform: [concatenatedInputSize, hiddenSize])
+        self.forgetWeight = Tensor<Scalar>(glorotUniform: [concatenatedInputSize, hiddenSize])
+        self.forgetBias = Tensor<Scalar>(zeros: [hiddenSize])
+        self.outputWeight = Tensor<Scalar>(glorotUniform: [concatenatedInputSize, hiddenSize])
+    }
+
+    public struct HiddenState: Differentiable {
```

Comment:
Regarding the naming of this, there are two issues: 
* It aliases `Output`, which can be confusing, though it's reasonable because it's technically correct.
* Type `HiddenState` does not meaningfully distinguish between a cell state and a hidden state (while it contains both as members!). Have you considered calling it just `State`?

Reply:
```swift
public struct State: Differentiable {
    var cell: Tensor<Scalar>
    var hidden: Tensor<Scalar>
}
```

Reply:
Done.